src/modules/verifiable-credentials/verifiable-credentials-base.service.ts

```diff
@@ -300,9 +308,33 @@ export abstract class VerifiableCredentialsServiceBase {
     const did = this._signerService.didHex;
 
     // TODO: generate types from the presentation
-    const types = verifiableCredential
-      ? verifiablePresentationWithCredentialEIP712Types
-      : verifiablePresentationEIP712Types;
+    const types: typeof verifiablePresentationWithCredentialEIP712Types =
+      JSON.parse(
+        JSON.stringify(verifiablePresentationWithCredentialEIP712Types)
+      );
+
+    const { oneVCHasExpirationDate, allVCHaveExpirationDate } =
```

Comment:
This can shortened to two lines if using `some` and `every` on `verifiableCredential`

Reply:
That’s true and with two loops instead of one as it is right now 😄